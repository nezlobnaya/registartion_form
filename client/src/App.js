import React from 'react';
import {BrowserRouter as Router, Route, NavLink, Switch} from 'react-router-dom';
import './App.css';
import Home from './components/Home/Home';
import RecordsList from './components/RecordsList/RecordsList';
import RecordSuccess from './components/Success/RecordSuccess';
import RecordAdd from './components/Form/Form';


function App() {
  return (
    <div className="App">     
      <Router>
        <Navigation />
        <div className="container">
          <Main />
        </div>
      </Router>
    </div>
  );
}

function Navigation() {
  return(
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark mb-4">
      <div className='container'>
        <ul className="navbar-nav mr-auto">
          <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/">Home</NavLink></li>
          <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/records/">Admin report</NavLink></li>
          <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/records/new">New Record</NavLink></li>
        </ul>
      </div>
    </nav>
  );
}

function Main() {
  return(
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/records" component={RecordsList} />
      <Route exact path="/records/new" component={RecordAdd} />
      <Route exact path="/success" component={RecordSuccess} />

    </Switch>
  );
}

export default App;