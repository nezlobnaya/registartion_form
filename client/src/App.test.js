import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  test('should render', () => {
    const { getByText } = render(<App />);
    const app = getByText('Per Aspera Ad Astra!!');
    expect(app).toBeInTheDocument();
  });

  test('Per Aspera Ad Astra!!', () => {
    render(<App />);
    const linkElement = screen.getByText(/Per Aspera Ad Astra!!/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('snapshot for App component', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('snapshot for App component with custom className', () => {
    const { container } = render(<App className="custom" />);
    expect(container).toMatchSnapshot();
  });

  test('snapshot for App component with custom style', () => {
    const { container } = render(<App style={{ background: 'red' }} />);
    expect(container).toMatchSnapshot();
  } );

  test('snapshot for App component with custom className and style', () => {  
    const { container } = render(<App className="custom" style={{ background: 'red' }} />);
    expect(container).toMatchSnapshot();
  } );

  test('snapshot for App component with custom className, style and children', () => {
    const { container } = render(<App className="custom" style={{ background: 'red' }}>
      <div>children</div>
    </App>);
    expect(container).toMatchSnapshot();
  } );

});



