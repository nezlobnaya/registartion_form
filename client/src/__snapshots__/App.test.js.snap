// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot for App component 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark mb-4"
    >
      <div
        class="container"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/"
            >
              Admin report
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/new"
            >
              New Record
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="jumbotron"
      >
        <h1>
          Per Aspera Ad Astra!!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> snapshot for App component with custom className 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark mb-4"
    >
      <div
        class="container"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/"
            >
              Admin report
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/new"
            >
              New Record
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="jumbotron"
      >
        <h1>
          Per Aspera Ad Astra!!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> snapshot for App component with custom className and style 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark mb-4"
    >
      <div
        class="container"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/"
            >
              Admin report
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/new"
            >
              New Record
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="jumbotron"
      >
        <h1>
          Per Aspera Ad Astra!!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> snapshot for App component with custom className, style and children 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark mb-4"
    >
      <div
        class="container"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/"
            >
              Admin report
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/new"
            >
              New Record
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="jumbotron"
      >
        <h1>
          Per Aspera Ad Astra!!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> snapshot for App component with custom style 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark mb-4"
    >
      <div
        class="container"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/"
            >
              Admin report
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/records/new"
            >
              New Record
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="container"
    >
      <div
        class="jumbotron"
      >
        <h1>
          Per Aspera Ad Astra!!
        </h1>
      </div>
    </div>
  </div>
</div>
`;
